plugins/utils/hardtime.nix

```diff
@@ -1,244 +1,220 @@
-        ];
-        "6" = [
-          "n"
-          "x"
-        ];
-        "7" = [
-          "n"
-          "x"
-        ];
-        "8" = [
-          "n"
-          "x"
-        ];
-        "9" = [
-          "n"
-          "x"
-        ];
-        "c" = [ "n" ];
-        "C" = [ "n" ];
-        "d" = [ "n" ];
-        "x" = [ "n" ];
-        "X" = [ "n" ];
-        "y" = [ "n" ];
-        "Y" = [ "n" ];
-        "p" = [ "n" ];
-        "P" = [ "n" ];
-      } "Keys in what modes that reset the count.";
-
-      restrictedKeys = helpers.defaultNullOpts.mkAttrsOf (with types; listOf str) {
-        "h" = [
-          "n"
-          "x"
-        ];
-        "j" = [
-          "n"
-          "x"
-        ];
-        "k" = [
-          "n"
-          "x"
-        ];
-        "l" = [
-          "n"
-          "x"
-        ];
-        "-" = [
-          "n"
-          "x"
-        ];
-        "+" = [
-          "n"
-          "x"
-        ];
-        "gj" = [
-          "n"
-          "x"
-        ];
-        "gk" = [
-          "n"
-          "x"
-        ];
-        "<CR>" = [
-          "n"
-          "x"
-        ];
-        "<C-M>" = [
-          "n"
-          "x"
-        ];
-        "<C-N>" = [
-          "n"
-          "x"
-        ];
-        "<C-P>" = [
-          "n"
-          "x"
-        ];
-      } "Keys in what modes triggering the count mechanism.";
-
-      restrictionMode =
-        helpers.defaultNullOpts.mkEnumFirstDefault
-          [
-            "block"
-            "hint"
-          ]
-          ''
-            The behavior when `restricted_keys` trigger count mechanism.
-          '';
-
-      disabledKeys = helpers.defaultNullOpts.mkAttrsOf (with types; listOf str) {
-        "<Up>" = [
-          ""
-          "i"
-        ];
-        "<Down>" = [
-          ""
-          "i"
-        ];
-        "<Left>" = [
-          ""
-          "i"
-        ];
-        "<Right>" = [
-          ""
-          "i"
-        ];
-      } "Keys in what modes are disabled.";
-
-      disabledFiletypes = helpers.defaultNullOpts.mkListOf types.str [
-        "qf"
-        "netrw"
-        "NvimTree"
-        "lazy"
-        "mason"
-      ] "`hardtime.nvim` is disabled under these filetypes.";
-
-      hints =
-        helpers.mkNullOrOption
-          (
-            with types;
-            attrsOf (submodule {
-              options = {
-                message = lib.mkOption {
-                  description = "Hint message to be displayed.";
-                  type = helpers.nixvimTypes.rawLua;
-                };
-
-                length = lib.mkOption {
-                  description = "The length of actual key strokes that matches this pattern.";
-                  type = types.ints.unsigned;
-                };
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "hardtime";
+  originalName = "hardtime.nvim";
+  package = "hardtime-nvim";
+
+  maintainers = [ lib.maintainers.refaelsh ];
+
+  deprecateExtraOptions = true;
```

Comment:
```suggestion
  # TODO: Added 2024-09-07; remove after 24.11
  deprecateExtraOptions = true;
```

Reply:
I've committed this change. But, can you please explain what it means?

Reply:
It's just for us to keep track of when to remove the deprecations so they become unavailable to use. We want deprecation warnings for a stable release before removing an option.